godtools/App/Features/SetupParallel/SetupParallelLanguageView.swift

```diff
@@ -45,63 +45,66 @@ class SetupParallelLanguageView: UIViewController {
         
         super.viewDidLoad()
         
+        tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleSelectLanguageTapped))
+        tapRecognizer.cancelsTouchesInView = false
+        tapRecognizer.delegate = self
+        self.view.addGestureRecognizer(tapRecognizer)
+        
         setupLayout()
         setupBinding()
     }
     
+    override func viewWillDisappear(_ animated: Bool) {
+        
+        self.view.removeGestureRecognizer(self.tapRecognizer)
```

Comment:
Is removing the gesture doing anything?  I think if it's removed it won't get added back because it is added on viewDidLoad.  If you do need to enable / disable a gesture that can also be done using the isEnabled property. https://developer.apple.com/documentation/uikit/uigesturerecognizer/1624220-isenabled

Reply:
I was just doing this for cleanup when the view gets closed.  I don't think I need to enable or disable it, I just want to properly remove the gesture recognizer when the screen gets closed.

Reply:
In that case you'd want to use the deinit to clean that up.  viewWillDisappear will get called if something is presented over this view, or if navigating forward to a view from a push etc. which would cause the tap gesture to get removed and then never added back.